packages/helia/src/utils/networked-storage.ts

```diff
@@ -71,8 +94,10 @@ export class NetworkedStorage implements Blocks {
     })
 
     const notifyEach = forEach(missingBlocks, ({ cid, block }): void => {
-      options.onProgress?.(new CustomProgressEvent<CID>('blocks:put-many:bitswap:notify', cid))
-      this.bitswap?.notify(cid, block, options)
+      options.onProgress?.(new CustomProgressEvent<CID>('blocks:put-many:providers:notify', cid))
+      this.blockProviders.forEach(provider => {
+        provider.notify(cid, block, options)
```

Comment:
```suggestion
        provider.notify?.(cid, block, options)
```

or better: keep a list of blockProviders that have `notify` methods and only call those.

Reply:
Done!